test/common.py

```diff
@@ -413,6 +419,33 @@ def accept_output(update_type):
             else:
                 self.assertEqual(s, expected)
 
+    # Runs `fn(*args, **kwargs)` with CUDA memory checks. Assumes that CUDA is
+    # available.
+    def run_with_cuda_memory_check(self, fn, *args, **kwargs):
+        num_devices = torch.cuda.device_count()
+        gc.collect()
+        torch.cuda.synchronize()
```

Comment:
Why do you perform first collect then synchronize? Is the order important or the two can't interact?

Reply:
The order shouldn't matter. As long as the refcnts go to zero it should be fine.